pkg/tekton/metapipeline/clientfactory.go

```diff
@@ -0,0 +1,295 @@
+package metapipeline
+
+import (
+	"fmt"
+	"github.com/jenkins-x/jx/pkg/apps"
+	"github.com/jenkins-x/jx/pkg/client/clientset/versioned"
+	jxclient "github.com/jenkins-x/jx/pkg/client/clientset/versioned"
+	"github.com/jenkins-x/jx/pkg/config"
+	"github.com/jenkins-x/jx/pkg/gits"
+	"github.com/jenkins-x/jx/pkg/jxfactory"
+	"github.com/jenkins-x/jx/pkg/kube"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/prow"
+	"github.com/jenkins-x/jx/pkg/tekton"
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	tektonclient "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	"io/ioutil"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"os"
+	"time"
+)
+
+const (
+	defaultContainerImage = "jx"
+	defaultServiceAccount = "tekton-bot"
+	retryDuration         = time.Second * 30
+	defaultCheckoutDir    = "source"
+)
+
+var (
+	logger = log.Logger().WithFields(logrus.Fields{"component": "meta-pipeline-client"})
+)
+
+// ClientFactory implements the interface methods to create and apply the meta pipeline.
+type ClientFactory struct {
+	jxClient         versioned.Interface
+	tektonClient     tektonclient.Interface
+	kubeClient       kubernetes.Interface
+	versionDir       string
+	ns               string
+	defaultImage     string
+	serviceAccount   string
+	versionStreamURL string
+	versionStreamRef string
+}
+
+// NewMetaPipelineClient creates a new client for the creation and application of meta pipelines.
+// The responsibility of the meta pipeline is to prepare the execution pipeline and to allow Apps to contribute
+// the this execution pipeline.
+func NewMetaPipelineClient() (*ClientFactory, error) {
```

Comment:
@ccojocar default impl of the interface. 

The `jx step create pipeline` is removed. So no command for this anymore.